engine.py

```diff
@@ -414,7 +506,7 @@ class Move():
     # map columns to files (revers of files to columns)
     cols_to_files = {col:file for file, col in files_to_cols.items()}
 
-    def __init__(self, start_sq, end_sq, board):
+    def __init__(self, start_sq, end_sq, board, is_castle_move = False):
```

Comment:
use move_type="normal" as default
and set move_type="castling" whenever you are creating a castling move.
This for consistency with the person working on en-passant move. So that we have one flag move_type to identify move types, instead of several flags (is_castle_move, en_passant, e.t.c)